src/Application/Regulation/Query/Location/GetRegulationLocationsQueryHandler.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Application\Regulation\Query\Location;
+
+use App\Application\Regulation\View\DetailLocationView;
+use App\Application\Regulation\View\RegulationOrderLocationsView;
+use App\Domain\Regulation\Exception\RegulationOrderRecordNotFoundException;
+use App\Domain\Regulation\RegulationOrderRecord;
+use App\Domain\Regulation\Repository\RegulationOrderRecordRepositoryInterface;
+
+final class GetRegulationLocationsQueryHandler
+{
+    public function __construct(
+        private RegulationOrderRecordRepositoryInterface $regulationOrderRecordRepository,
+    ) {
+    }
+
+    public function __invoke(GetRegulationLocationsQuery $query): RegulationOrderLocationsView
+    {
+        $regulationOrderRecord = $this->regulationOrderRecordRepository->findOneForSummary(
+            $query->uuid,
+        );
+
+        if (!$regulationOrderRecord instanceof RegulationOrderRecord) {
+            throw new RegulationOrderRecordNotFoundException();
+        }
+
+        $regulationOrder = $regulationOrderRecord->getRegulationOrder();
+        $locationViews = [];
+
+        foreach ($regulationOrder->getLocations() as $location) {
+            $locationViews[] = DetailLocationView::fromEntity($location);
+        }
```

Comment:
On pourrait faire tout ça plutôt en se "centrant" sur Location :

```suggestion
        $locations = $this->locationRepository->findAllByRegulationOrderRecordUuid($query->uuid);
```

Ça minimise de deux crans le nombre de jointures (car pour RegulationOrderRecord et RegulationOrder on peut juste faire une jointure par ID pour aller faire le `WHERE roc.uuid = :uuid`)

Et pour la nouvelle UI d'affichage on devrait avoir une requête tout à fait analogue "centrée" sur les mesures :

```php
        $measures = $this->measureRepository->findAllByRegulationOrderRecordUuid($query->uuid);
```

Qui pourait renvoyer les données des `locationNew` et (dans un premier temps) ne passer par `Location` que pour remonter jusqu'à `roc`

Ce qui évite de devoir brancher RegulationOrder et Measure juste pour implémenter l'UI d'affichage, non ? Car sinon je ne sais pas comment tu as fais pour la nouvelle UI, à moins d'avoir fait les 2 (affichage et édition) en même temps, par tentacularité

Reply:
OK !